### PR TITLE
add option loginUrl

### PR DESCRIPTION
**Pull request by GiuseppeCM**
add option loginUrl to change the default login url.

### CODE REVIEW

File: lib/modules/apostrophe-login/index.js [hunk 1]
```diff
@@ -349,12 +353,24 @@ module.exports = {
       return next();
     };
 
+    // Check if the first character about the option `loginUrl` is `/`
+    // and if is not present, add the `/` in the first position.
+    // If the option is not present, return the default url `/login`
+    self.fetchLoginUrl = function() {
+      if (self.options.loginUrl) {
+        return self.options.loginUrl && self.options.loginUrl.startsWith('/') ? self.options.loginUrl : `/${self.options.loginUrl}`
```
Comment (boutell) [inline lib/modules/apostrophe-login/index.js]: We should skip this mechanism, the developer is responsible for including the leading / on all other URL based options and I don't want to cause any confusion about that.

File: lib/modules/apostrophe-login/index.js [hunk 2]
```diff
@@ -349,12 +353,24 @@ module.exports = {
...
+      }
+    }
+
     // Add the `/login` route, both GET (show the form) and POST (submit the form).
     // Also add the `/logout` route.
 
     self.addRoutes = function() {
       if (self.options.localLogin !== false) {
-        self.apos.app.get('/login', function(req, res) {
+        self.apos.app.get(self.fetchLoginUrl(), function(req, res) {
```
Comment (boutell) [inline lib/modules/apostrophe-login/index.js]: This URL appears in many other places: for CSRF protection, for redirects from other steps and for other things. You will need to switch the code to call getLoginUrl in each case.

File: lib/modules/apostrophe-login/index.js [hunk 3]
```diff
@@ -349,12 +353,24 @@ module.exports = {
       return next();
     };
 
+    // Check if the first character about the option `loginUrl` is `/`
+    // and if is not present, add the `/` in the first position.
+    // If the option is not present, return the default url `/login`
+    self.fetchLoginUrl = function() {
```
Comment (boutell) [inline lib/modules/apostrophe-login/index.js]: getLoginUrl would be more consistent with the rest of the application.

File: lib/modules/apostrophe-login/views/passwordResetRequest.html [hunk 4]
```diff
@@ -26,7 +26,7 @@
               <input type="submit" class="apos-button apos-button--major" value="{% block buttonLabel %}{{ __('Request Password Reset') }}{% endblock %}" />
             </div>
             <div class="apos-login-reset-cancel">
-              <a href="{{ apos.prefix }}/login">{{ __('Cancel') }}</a>
+              <a href="{{ apos.prefix }}{{data.loginUrl}}">{{ __('Cancel') }}</a>
```
Comment (boutell) [inline lib/modules/apostrophe-login/views/passwordResetRequest.html]: Where is this getting into the template data?

Comment (GiuseppeCM) [reply lib/modules/apostrophe-login/views/passwordResetRequest.html]: https://github.com/apostrophecms/apostrophe/pull/2124/commits/32337ca76f483931dd45935611858115cff82fa6#diff-297b2881cc29c9a42671c4bc64f95986R454 this?

**Comment by boutell** [comment]
Please fix eslint issues:

```
npx eslint . --fix
```

And manually address anything that points out as not fixable automatically.

**Pull request merged**